frontend/src/views/welcome.js

```diff
@@ -1,30 +1,56 @@
 import React from 'react';
 import { useSelector } from 'react-redux';
-import { redirectTo } from '@reach/router';
+import { Redirect } from '@reach/router';
 
 import { UserTopBar } from '../components/user/settings';
-import { HelpCard } from '../components/user/help';
-import { UserInformationForm, UserInterests } from '../components/user/forms';
+import { HelpCard, FirstProjectBanner } from '../components/user/content';
+import { calculateCompleteness } from '../components/user/completeness';
+import { UserInformationForm } from '../components/user/forms';
+import { WelcomeCard } from '../components/user/content';
+
+
+function IncompleteProfile() {
+  return (
+    <>
+      <UserTopBar />
+      <div className="cf pa4 bg-tan">
+        <div className="fl w-100 w-60-l pb3 pr3-l">
+          <WelcomeCard />
+          <HelpCard />
+        </div>
+        <div className="fl w-100 w-40-l pb3 pl3-l">
+          <UserInformationForm />
+        </div>
+      </div>
+    </>
+  );
+}
+
+function NewContributor() {
+  return (
+    <div className="cf pa4 bg-tan">
+      <div className="fl w-100 pb3">
+        <FirstProjectBanner />
+        <p>Space for recommended projects</p>
+      </div>
+    </div>
+  );
+}
 
 export function Welcome() {
+  const userDetails = useSelector(state => state.auth.get('userDetails'));
   const userIsloggedIn = useSelector(state => state.auth.get('token'));
 
   if (userIsloggedIn) {
+    const completeness = calculateCompleteness(userDetails);
     return (
       <div className="pull-center">
-        <UserTopBar />
-        <div className="cf pa4 bg-tan">
-          <div className="fl w-100 w-60-l pb3 pr3-l">
-            <UserInterests />
-            <HelpCard />
-          </div>
-          <div className="fl w-100 w-40-l pb3 pl3-l">
-            <UserInformationForm />
-          </div>
-        </div>
+        {completeness <= 0.5 ? <IncompleteProfile />
```

Comment:
See wrapping with `<react-placeholder>` comment above.

Reply:
This one was jumping from 0->90% too as the data loaded. Maybe you fixed it somehow, when I get a chance to try this tomorrow I will check it.